core/services/relay/evm/chain_reader.go

```diff
@@ -3,14 +3,19 @@ package evm
 import (
 	"context"
 	"fmt"
-	"slices"
```

Comment:
I feel like methodName is overloaded and typeName is too general so I made a PR that should make chainreader easier to understand https://github.com/smartcontractkit/chainlink/pull/11561/files. .
This is the same PR that I mentioned for bindings